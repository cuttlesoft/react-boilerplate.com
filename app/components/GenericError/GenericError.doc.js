import { describe, PropTypes } from 'react-desc'

export const doc = GenericError => {
  const DocumentedGenericError = describe(GenericError)
    .description('A GenericError')
    .usage(
      `import { GenericError } from 'components/GenericError';
<GenericError />`,
    )

  DocumentedGenericError.propTypes = {
    id: PropTypes.string
      .description('The DOM id attribute value to use for the element.')
      .defaultValue(false),
  }

  return DocumentedGenericError
}
