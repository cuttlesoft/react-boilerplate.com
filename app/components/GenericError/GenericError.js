import React from 'react'

import { Box } from '../Box'
import { Heading } from '../Heading'

/**
 * GenericError
 */
const GenericError = () => (
  <Box height="90vh" justify="center">
    <Heading level="1" margin={{ bottom: 'small' }} size="large" textAlign="center">
      <span role="img" aria-label="Crying Face">
        ðŸ˜¢
      </span>
    </Heading>

    <Heading level="2" margin={{ top: 'small' }} textAlign="center">
      Could not load data. Try refreshing your page.
    </Heading>
  </Box>
)

export default GenericError
