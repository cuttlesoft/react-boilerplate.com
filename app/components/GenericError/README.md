<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## GenericError
A GenericError

## Usage

```javascript
import { GenericError } from 'components/GenericError';
<GenericError />
```

## Properties

**id**

The DOM id attribute value to use for the element.

```
string
```
  