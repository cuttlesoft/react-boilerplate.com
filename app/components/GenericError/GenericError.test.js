import React from 'react'
import { render } from '@testing-library/react'
// import '@testing-library/jest-dom/extend-expect' // add some helpful assertions

import GenericError from './GenericError'

const renderComponent = (props = {}) => render(<GenericError {...props} />)

/**
 *
 * Tests for GenericError
 *
 */
describe('GenericError', () => {
  it('renders and matches snapshot', () => {
    const { container } = renderComponent()
    expect(container).toMatchSnapshot()
  })
})
