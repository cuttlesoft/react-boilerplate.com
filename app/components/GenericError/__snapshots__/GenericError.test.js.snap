// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericError renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 kibWqE"
  >
    <h1
      class="StyledHeading-sc-1rdh4aw-0 bgWVVT"
      style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
    >
      <span
        aria-label="Crying Face"
        role="img"
      >
        ðŸ˜¢
      </span>
    </h1>
    <h2
      class="StyledHeading-sc-1rdh4aw-0 ghwJhK"
      style="font-family: soleil, 'Helvetica Neue', Helvetica, Arial, sans-serif; max-width: 100vw; text-transform: false;"
    >
      Could not load data. Try refreshing your page.
    </h2>
  </div>
</div>
`;
